src/core/ActivityInfoTree.cpp

```diff
@@ -283,5 +285,71 @@ void ActivityInfoTree::init()
 
     qmlRegisterSingletonType<QObject>("GCompris", 1, 0, "ActivityInfoTree", menuTreeProvider);
     qmlRegisterType<ActivityInfo>("GCompris", 1, 0, "ActivityInfo");
+
+
+}
+void ActivityInfoTree::filterBySearch(const QString& text)
+{
+
+    if(text.trimmed().length() < m_searchedtext.length() or text.trimmed() == ""){
```

Comment:
If I remember, we said it was not useful to optimize?
It causes also issues if when you are in a search: let's say you typed "chess", you select the 'e' and type 'a'. No activities should match but in your case, nothing will change